snuba/datasets/groups.py

```diff
@@ -58,12 +36,9 @@ class Groups(TimeSeriesDataset):
     GROUPS_ALIAS = "groups"
 
     def __init__(self) -> None:
-        self.__grouped_message = get_dataset("groupedmessage")
         groupedmessage_source = (
             groupedmessages_storage.get_schemas().get_read_schema().get_data_source()
         )
-
-        self.__events = get_dataset("events")
         events_source = events_storage.get_schemas().get_read_schema().get_data_source()
 
         join_structure = JoinClause(
```

Comment:
I took a bit of shortcut here.
JoinClause is a RelationalSource. It is what we set in the Query and that is able to format the FROM clause with the join in the clause. We may say this should be produced on the fly by JoinQueryPlanBuilder based on a more abstract specification provided by the dataset when processing the query. In practice as of today such higher level abstraction and JoinClause would be identical.
I think this problem should be revisited when we the JoinQueryPlanBuilder will receive a join definition expressed in terms of Entities. Then JoinClause will not be suitable anymore and we will replace it.